### ISSUE TITLE
Transpile: sqlite to postgres, AUTOINCREMENT columns don't transpile quite right

### ISSUE THREAD

**Issue by adamf** [tobymao/sqlglot#464]
Given sqlite:
```
CREATE TABLE people (
   person_id INTEGER PRIMARY KEY AUTOINCREMENT,
)
```

The (incorrect) transpile to postgres is:
```
CREATE TABLE "people" (
        person_id INT PRIMARY KEY SERIAL
);
```

The correct would be:
```
CREATE TABLE "people" (
       person_id SERIAL PRIMARY KEY
);
```

**Comment by mpf82** [comment]
@tobymao Sorry, but this fix generates invalid postgres statements.

``CREATE TABLE z (a INT UNIQUE SERIAL PRIMARY KEY)`` should be 

``CREATE TABLE z (a SERIAL unique primary key)``

``serial`` is a datatype that comes with an autoincrement feature, you cannot combine it with other datatypes, such as ``int``

Since autoincrement translates to SERIAL and vice versa, there's another issue: the "precision" is lost, because there are actually 3 autoincrement types in postgres: SMALLSERIAL, SERIAL, and BIGSERIAL

See: https://www.postgresql.org/docs/current/datatype-numeric.html

related issue: https://github.com/tobymao/sqlglot/issues/301

**Comment by tobymao** [comment]
fixed in 8b3efca
